### DISCUSSION TITLE
Auto login when multiple providers

### DISCUSSION THREAD

**Discussion by ppa-xyt** [outline/outline#2648] [Ideas]
We have enabled Outline authentication via two Identity Providers. Some users should always use one IdP to log in, while the others should use the second one. 

When logging in to Outline, user has to click the button to choose the proper IdP. Are there any options to skip this step?

The idea is that Outline keeps information (cookie?) how did user log in last time. Is it possible to leverage this information and authenticate user using last chosen IdP?

**Comment by almereyda** [comment]
When I login to Outline, my access token cookie is valid for a month.

Which is the grace period when it will be renewed? I don't know.

Another remotely related way to implement this kind of Single-Sign On scenario could be through SAML support, where this can be managed at the IAM.
